arangod/VocBase/Properties/UserInputCollectionProperties.cpp

```diff
@@ -50,19 +50,8 @@ UserInputCollectionProperties::applyDefaultsAndValidateDatabaseConfiguration(
   // into ClusteringProperties. Doing it here was the quicker to implement way,
   // so we went with it first DistributeShardsLike has the strongest binding. We
   // have to handle this first
-  if (!config.defaultDistributeShardsLike.empty()) {
-    if (!distributeShardsLike.has_value()) {
-      distributeShardsLike = config.defaultDistributeShardsLike;
-    } else {
-      // NOTE: For the time being only oneShardDBs have a default
-      // distributeShardsLikeValue. So this error message is good enough. If
-      // this should ever change we need to adapt the message here. NOTE: The
-      // assertion is a reminder to update the following error message.
-      TRI_ASSERT(config.isOneShardDB);
-      return {TRI_ERROR_BAD_PARAMETER,
-              "Collection in a 'oneShardDatabase' cannot define "
-              "distributeShardsLike"};
-    }
+  if (!config.defaultDistributeShardsLike.empty() && !distributeShardsLike.has_value()) {
```

Comment:
Note te reviewer:
There is already a check for oneShard and distributeShardsLike
here:
https://github.com/arangodb/arangodb/blob/bug-fix/allow-oneshard-with-matching-distributeshardslike/arangod/VocBase/Properties/ClusteringConstantProperties.cpp#L57

This entry was more or less a duplicate and the two variants diverged. The linked variant does the test in the intended way, so erasing this actually removed the false negative.